regen/feature.pl

```diff
@@ -491,6 +484,49 @@ sub longest {
 }
 #endif /* PERL_IN_MG_C */
 
+/* subject to change */
+struct perl_feature_bit {
+  const char *name;
+  STRLEN namelen;
+  U32 mask;
+};
+
+#ifdef PERL_IN_PP_CTL_C
+
+static const struct perl_feature_bit
+PL_feature_bits[] = {
+EOJ
+for my $key (sort keys %feature) {
+    my $val = $feature{$key};
+    print $h <<EOJ;
+    {
+        /* feature $key */
+        "feature_$val",
+        STRLENs("feature_$val"),
+        FEATURE_\U$val\E_BIT
+    },
+EOJ
+}
+
+print $h <<EOJ;
+    { NULL, 0, 0U }
+};
+
+PERL_STATIC_INLINE void
+S_fetch_feature_bits_hh(pTHX_ HV *hh) {
+    PL_compiling.cop_features = 0;
+
+    const struct perl_feature_bit *fb = PL_feature_bits;
+    while (fb->name) {
+        SV **svp = hv_fetch(hh, fb->name, (I32)fb->namelen, 0);
+        if (svp && SvTRUE(*svp))
+               PL_compiling.cop_features |= fb->mask;
+        ++fb;
```

Comment:
This while loop has 26 hash key name entries to test/look for. All 26 hash keys, are never found in typical production perl code. The `HV*` called `hh`, has only 1 HE*/key in it total. O(n) problems here.